pkg/controller/provider/container/ova/model.go

```diff
@@ -228,3 +231,36 @@ func (r *DiskAdapter) GetUpdates(ctx *Context) (updates []Updater, err error) {
 	}
 	return
 }
+
+type StorageAdapter struct {
+	BaseAdapter
+}
+
+func (r *StorageAdapter) GetUpdates(ctx *Context) (updates []Updater, err error) {
+	return
+}
+
+// List the collection.
+func (r *StorageAdapter) List(ctx *Context, provider *api.Provider) (itr fb.Iterator, err error) {
+	storageList := []Storage{}
+	storageName := fmt.Sprintf("Provider %s storage", provider.Name)
+	storageList = append(storageList, Storage{
+		Name: storageName,
+		ID:   string(provider.UID),
+	})
+	list := fb.NewList()
+	for _, object := range storageList {
+		m := &model.Storage{
+			Base: model.Base{
+				ID:   object.ID,
+				Name: object.Name,
+			},
+		}
+		object.ApplyTo(m)
+		list.Append(m)
+	}
```

Comment:
I understand that the code above is based on similar code we have for other providers but considering that we have a single storage, how about simplifying it?

Reply:
after looking into it, all the adapters must implement the same interface and return the iterator from the func 
(which needed for the next func under the collector) since its  exists only for the list object I cant change the implementation even thought we having only one resource here.

Reply:
I'm not suggesting to change the signature but creating a list `storageList` that includes a single value and then iterating over it.. would the following work instead:
```golang
storageName := fmt.Sprintf("Provider %s storage", provider.Name)
list := fb.NewList()
m := &model.Storage{
	Base: model.Base{
		ID:   string(provider.UID),
		Name: storageName,
	},
}
list.Append(m)
itr = list.Iter()
return
```
?